clients/java/src/main/java/io/camunda/zeebe/client/impl/search/ProcessInstanceSortImpl.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.impl.search;
+
+import io.camunda.zeebe.client.api.search.ProcessInstanceSort;
+
+public class ProcessInstanceSortImpl extends SearchQuerySortBase<ProcessInstanceSort>
+    implements ProcessInstanceSort {
+
+  @Override
+  public ProcessInstanceSort startDate() {
+    return field("startDate");
```

Comment:
[❓If I am not wrong, on Camunda Services, we refer those fields to sort, why we need re-do it here on the `SortImpl`?

Reply:
Camunda Services is a server-side implementation, and the sorting there defines on which (index) field sorting must happen.

This code here is used on the client side, which eventually calls the `/process-instances/search` endpoint. At this point, the user constructs the search query being serialized into a JSON, for example:
```json
{
  "filter": {
    "processInstanceKey": 22456786958
  },
  "sort": [
    { "startDate": "ASC" }
  ],
  "page": {
    "limit": 10
  }
}
```
When the Process Instance REST controller receives that request, it must somehow map the client's search request to the actual service, i.e., based on the passed sorting field it will call the respective sort method in the service.